Assets/uAdventure/__Scripts/Runner/GameLogic/GUIManager.cs

```diff
@@ -373,53 +375,72 @@ public void ClearData()
 
         public void ExitApplication()
         {
-            if (PlayerPrefs.HasKey("LimesurveyToken") && PlayerPrefs.GetString("LimesurveyToken") != "ADMIN" && PlayerPrefs.HasKey("LimesurveyPost"))
+            if (SimvaController.Instance && SimvaController.Instance.isActive())
             {
-                string path = Application.persistentDataPath;
-
-                if (!path.EndsWith("/"))
+                JSONNode activity = SimvaController.Instance.getActivity(SimvaController.Instance.getCurrentActivityId());
+                string activityType = activity["type"].Value;
+                if (activityType == "miniokafka" || activityType == "activity")
                 {
-                    path += "/";
-                }
-
-                Dictionary<string, string> headers = new Dictionary<string, string>();
-
-                Net net = new Net(this);
-
-                WWWForm data = new WWWForm();
+                    string path = Application.persistentDataPath;
+                    if (!path.EndsWith("/"))
+                    {
+                        path += "/";
+                    }
 
-                TrackerAssetSettings trackersettings = (TrackerAssetSettings)TrackerAsset.Instance.Settings;
-                string backupfile = Application.persistentDataPath + System.IO.Path.DirectorySeparatorChar + trackersettings.BackupFile;
+                    TrackerAssetSettings trackersettings = (TrackerAssetSettings)TrackerAsset.Instance.Settings;
+                    string backupfile = path + trackersettings.BackupFile;
+                    string traces = System.IO.File.ReadAllText(backupfile);
+                    System.IO.File.AppendAllText(path + SimvaController.Instance.Token + ".csv", traces);
 
-                data.AddField("token", PlayerPrefs.GetString("LimesurveyToken"));
-                data.AddBinaryData("traces", System.Text.Encoding.UTF8.GetBytes(System.IO.File.ReadAllText(backupfile)));
+                    StartCoroutine(SaveActivityAndContinue(SimvaController.Instance.getCurrentActivityId(), traces, true));
+                }
+                else
+                {
+                    StartCoroutine(Continue(SimvaController.Instance.getCurrentActivityId(), true));
+                }
+            }
+            else
+            {
+                Application.Quit();
+            }
```

Comment:
I still think this code should belong to analyticsextension.cs and should be plugged to the game.cs lifecycle any form

Reply:
Fixed by adding a OnGameFinished method to all extensions preventing the game to be closed and hooking it